espnet/bin/asrtts_train.py

```diff
@@ -0,0 +1,439 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+
+# Copyright 2017 Tomoki Hayashi (Nagoya University)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+import configargparse
+import logging
+import os
+import platform
+import random
+import subprocess
+import sys
+
+import numpy as np
+
+from espnet.utils.cli_utils import strtobool
+from espnet.utils.training.batchfy import BATCH_COUNT_CHOICES
+
+
+# NOTE: you need this func to generate our sphinx doc
+def get_parser():
+    parser = configargparse.ArgumentParser(
+        description="Train an automatic speech recognition (ASR) model on one CPU, one or multiple GPUs",
+        config_file_parser_class=configargparse.YAMLConfigFileParser,
+        formatter_class=configargparse.ArgumentDefaultsHelpFormatter)
+    # general configuration
+    parser.add('--config', is_config_file=True, help='config file path')
+    parser.add('--config2', is_config_file=True,
+               help='second config file path that overwrites the settings in `--config`.')
+    parser.add('--config3', is_config_file=True,
+               help='third config file path that overwrites the settings in `--config` and `--config2`.')
+
+    parser.add_argument('--ngpu', default=0, type=int,
+                        help='Number of GPUs')
+    parser.add_argument('--backend', default='chainer', type=str,
+                        choices=['chainer', 'pytorch'],
+                        help='Backend library')
+    parser.add_argument('--outdir', type=str, required=True,
+                        help='Output directory')
+    parser.add_argument('--debugmode', default=1, type=int,
+                        help='Debugmode')
+    parser.add_argument('--dict', required=True,
+                        help='Dictionary')
+    parser.add_argument('--seed', default=1, type=int,
+                        help='Random seed')
+    parser.add_argument('--debugdir', type=str,
+                        help='Output directory for debugging')
+    parser.add_argument('--resume', '-r', default='', nargs='?',
+                        help='Resume the training from snapshot')
+    parser.add_argument('--minibatches', '-N', type=int, default='-1',
+                        help='Process only N minibatches (for debug)')
+    parser.add_argument('--verbose', '-V', default=0, type=int,
+                        help='Verbose option')
+    parser.add_argument('--tensorboard-dir', default=None, type=str, nargs='?', help="Tensorboard log dir path")
+    # task related
+    parser.add_argument('--train-json', type=str, default=None, nargs='+',
```

Comment:
@creatorscan 
Did you make codes for using `args.train_json` of list type?

Error abut data-type was happened.
In ASR: `args.train_json` may be str.
In ASRTTS: `args.train_json` is list.
```
# asrtts_train.py --config conf/train_asrtts.yaml --ngpu 1 --backend pytorch --outdir exp/asrtts__exploss_pgrad/results --debugmode 1 --dict data/lang_char/train_960_units.txt --debugdir exp/asrtts__exploss_pgrad --minibatches 0 --verbose 0 --resume --train-json dump/train_clean_360/data_rnd.json dump/train_clean_100/data.json --valid-json dump/dev/data.json --opt adadelta --asr-model-conf exp/asr_/results/model.json --asr-model exp/asr_/results/model.acc.best --tts-model-conf exp/tts_/results/model.json --tts-model exp/tts_/results/model.loss.best --update-asr-only --policy-gradient 
# Started at Thu Sep 19 04:36:14 JST 2019
#
2019-09-19 04:36:15,845 (asrtts_train:376) WARNING: Skip DEBUG/INFO messages
2019-09-19 04:36:15,855 (asrtts_train:394) WARNING: CUDA_VISIBLE_DEVICES is not set.
2019-09-19 04:36:21,007 (e2e_asr:90) WARNING: Subsampling is not performed for vgg*. It is performed in max pooling layers at CNN.
Traceback (most recent call last):
  File "/work/abelab/k_inoue/debug/espnet/egs/librispeech/asrtts/../../../espnet/bin/asrtts_train.py", line 439, in <module>
    main(sys.argv[1:])
  File "/work/abelab/k_inoue/debug/espnet/egs/librispeech/asrtts/../../../espnet/bin/asrtts_train.py", line 427, in main
    train(args)
  File "/work/abelab/k_inoue/debug/espnet/espnet/asr/pytorch_backend/asr.py", line 356, in train
    with open(args.train_json, 'rb') as f:
TypeError: expected str, bytes or os.PathLike object, not list
# Accounting: time=12 threads=1
# Ended (code 1) at Thu Sep 19 04:36:26 JST 2019, elapsed time 12 seconds
```